### ISSUE TITLE
Proposal for improved Table Element

### ISSUE THREAD

**Issue by philippjfr** [ioam/holoviews#225]
In some recent discussions I mentioned that ``Table`` was not designed to hold large tables (>10k cells) as its based on ``NdMapping``, which is really designed to hold individual elements. Any real data manipulation on a Table Element can become very slow. Since ``Table`` elements are the common interchange format when performing many operations, e.g. sample, reduce, this can happen pretty quickly.

My suggestion is as follows, much like we have different plotting backends we could have different data backends. The implementation of ``DFrame`` already provides most of the functionality that ``Table`` provides but allows for much faster operations and conversions. Instead of having two separate classes the user should be able to decide which implementation of ``Table`` HoloViews should use internally, e.g. inbuilt, pandas or xray. As long as the different ``Table`` implementations share the same public API this would be a clean and very easy (the current ``DFrame`` could be made to match the ``Table`` API fairly easily) way of getting a more performant ``Table`` implementation.

I'd also consider deprecating the ``dframe`` method since you could get a pandas based ``Table`` object by calling ``.table(backend='pandas')`` or switching the global data backend to 'pandas'.

I'm not 100% decided on implementation details but I guess we'd just have multiple classes called ``Table`` for each data backend. I'd be very happy to hear other suggestions though.

**Comment by jbednar** [comment]
That all sounds good to me, and very important to do.   

BTW, recently I've been using blaze to read files in, but then converting them to Pandas dataframes and then to Numpy arrays: 

http://continuumio.github.io/datashader/gdelt.html
http://continuumio.github.io/datashader/nyc-taxi.html

It would be better to leave them directly as Blaze objects or at least Pandas objects, if that would work properly, because then people could have low-cost visualization of what they are working with, without having to move everything into HoloViews objects that duplicate the memory usage.  

If there's a better way to do this already, let me know!  E.g. I wasn't sure how to use Pandas without Seaborn, as the examples I saw all involved Seaborn, so I just converted it to Numpy arrays instead.
